File: bpf_prog_load_kprobe.py
from bcc import BPF

# BPF program
bpf_program = """
int trace_bpf_prog_load(struct pt_regs *ctx) {
    bpf_trace_printk("bpf_prog_load called\\n");
    return 0;
}
"""

# Load BPF program
b = BPF(text=bpf_program)

# Attach kprobe to bpf_prog_load
b.attach_kprobe(event="bpf_prog_load", fn_name="trace_bpf_prog_load")

print("Tracing bpf_prog_load... Ctrl+C to exit.")

# Print trace output
b.trace_print()

File: bpf_prog_load_kprobe.c
#include <linux/bpf.h>
#include <bpf/bpf_helpers.h>

SEC("kprobe/bpf_prog_load")
int bpf_prog_load_hook(struct pt_regs *ctx) {
    bpf_printk("bpf_prog_load called\n");
    return 0;
}

char LICENSE[] SEC("license") = "GPL";
